ecl/hql/hqlgram.y

```diff
@@ -7976,27 +7976,29 @@ simpleDataSet
                             $$.setExpr(createDataset(no_choosesets, $3.getExpr(), $5.getExpr()));
                             $$.setPosition($1);
                         }
-    | DEDUP '(' startTopLeftRightSeqFilter optDedupFlags ')' endTopLeftRightFilter endSelectorSequence
+    | DEDUP '(' startSortOrder startTopLeftRightSeqFilter optDedupFlags ')' endTopLeftRightFilter endSelectorSequence endSortOrder
```

Comment:
@jakesmith I did think about implementing it as an expression but not sure how it would it be handled if all expressions were calculated as false- e.g. if in the above case all field2<10 and field3>10.   What if for some rows, the first expressions was true & second was false; and for others vice-versa?   Also, Gavin's jira ticket suggested the syntax as BEST(<sort-list>)... I'd like more clarification on this.,